crates/zk-por-core/src/bin/prover.rs

```diff
@@ -106,23 +125,60 @@ fn main() {
             "parse {} times, with number of accounts {}, number of batches {}",
             parse_num,
             account_num,
-            account_num / batch_size
+            expected_batch_num,
         );
 
+        let batch_idx_base = batch_proofs.len();
+        let root_hashes: Vec<HashOut<F>> = accounts
+            .par_chunks(batch_size)
+            .enumerate()
+            .map(|(i, account_batch)| {
+                let batch_idx = batch_idx_base + i;
+                let mst = MerkleSumTree::new_tree_from_accounts(&account_batch.to_vec());
+
+                let global_mst = GLOBAL_MST.get().unwrap();
+                let mut _g = global_mst.write().expect("unable to get a lock");
+
+                for inner_tree_node_idx in 0..batch_size * 2 - 1 {
+                    _g.set_batch_hash(
+                        batch_idx,
+                        inner_tree_node_idx,
+                        mst.merkle_sum_tree[inner_tree_node_idx].hash,
+                    );
+                }
+                drop(_g);
+                mst.get_root().hash
+            })
+            .collect();
+
         let proofs = batch_prove_accounts(
             &circuit_registry,
             accounts,
             BATCH_PROVING_THREADS_NUM,
             batch_size,
         );
-        offset += per_parse_account_num;
+
+        assert_eq!(proofs.len(), root_hashes.len());
+
+        proofs.iter().zip(root_hashes.iter()).enumerate().for_each(|(i, (proof, root_hash))|{
+            let batch_idx = batch_idx_base + i;
+            // exclude the first two pub inputs for equity and debt
+            let proof_root_hash = HashOut::<F>::from_partial(&proof.public_inputs[2..]);
```

Comment:
[suggestion] can put the offset to constant; more flexible in management if the public inputs change in the future